components/encryption/src/log_file.rs

```diff
@@ -153,69 +151,64 @@ impl LogFile {
                 }
             }
         }
+        self.file_dict = file_dict.clone();
         Ok(file_dict)
     }
 
     /// Append an insert operation to the log file.
     ///
     /// Warning: `self.write(file_dict)` must be called before.
     pub fn insert(&mut self, name: &str, info: FileInfo) -> Result<()> {
-        if let Some(file) = self.append_file.as_mut() {
-            let bytes = Self::convert_record_to_bytes(name, LogRecord::INSERT(info))?;
-            file.write_all(&bytes)?;
-            file.sync_all()?;
-            self.check_compact()?;
-            Ok(())
-        } else {
-            Err(box_err!(
-                "file corrupted! write operation must be called before append."
-            ))
-        }
+        let file = self.append_file.as_mut().unwrap();
+        let bytes = Self::convert_record_to_bytes(name, LogRecord::INSERT(info.clone()))?;
```

Comment:
Is it possible or necessary to avoid to clone `FileInfo`?

Reply:
It will be more troublesome if use references.